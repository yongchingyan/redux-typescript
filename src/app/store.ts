import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import displayReducer from '../features/display/displaySlice';

export const store = configureStore({
  reducer: {
    display: displayReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
