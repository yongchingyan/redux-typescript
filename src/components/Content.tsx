import React, { Component } from 'react';
import '../css/content.scss'
import data  from "../asset/assets.json"

interface AssetData {
    imagePath: string;
    format: string;
    link: string;
    content: Content
    image?: string;
}

interface Content {
    button?: string;
    overlay?: string;
    text?: string;
    prompt?:  string;
}

interface IProps {
}

interface IState {
  assetsStore?: AssetData[];
}

export class MainContent extends Component<IProps, IState> {
    
    constructor(props: IProps) {
        super(props)
        this.state = {
            assetsStore: []
        }
    }

    componentDidMount() {
        // this.setState({assetsStore: []})
        let assetsStore: any[] = []
        data.assets.forEach((asset, key) => {
            import(`../asset/${asset.imagePath}`)
            .then(image=>{
                assetsStore[key] = {...asset, image: image.default}
                this.setState({assetsStore})
            });
        });
    }

    render() {

        let content; 
        if (this.state.assetsStore) {
            content = this.state.assetsStore.map(asset => {

                let temp: JSX.Element[]  = []
                let content : Content = asset.content
                if (Object.keys(content).length > 0){
                    
                    if ("text" in content) {
                        temp.push(
                            <h3 key={`text-${asset.imagePath}`}>
                                {content["text"]}
                            </h3>
                        )
                    }
                    if ("button" in content) {
                        temp.push(
                            <button key={`button-${asset.imagePath}`}>
                                {content["button"]}
                            </button>
                        )
                    }
                    if ("prompt" in content){
                        // temp.push(<a className="prompt" href={asset.link}>{content["prompt"]}</a>)
                        temp.push(
                            <span className="prompt" key={`prompt-${asset.imagePath}`}>
                                {content["prompt"]}
                            </span>)
                    }
                    if ("overlay" in content){
                        temp.push(
                            <div className="overlay" key={`overlay-${asset.imagePath}`}></div>
                        )
                    }
                }
                
                return (
                    <div key={`div-${asset.imagePath}`} className={asset.format}>
                        <a key={`a-${asset.imagePath}`} className="clickable-div" href={asset.link} >
                            <React.Fragment >
                                {temp}  
                            </React.Fragment>
                            <img key={`img-${asset.imagePath}`} alt="Game Banner" src={asset.image}/> 
                        </a>
                    </div> 
                )
            })
        }

        const {assetsStore} = this.state
        return (
            <div id="content">
                {/* <img src={imageSrc} /> */}
                <div className="wrapper">
                    { assetsStore && assetsStore.length > 0 ? 
                        content
                     : null}
                </div>
            </div>
        )
    }
}