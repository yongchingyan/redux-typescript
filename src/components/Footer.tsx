import React from 'react'
import '../css/footer.scss'
import logoSmall from '../asset/logo-small.png'

export const Footer = () => {
    return(
        <footer id="footer">
            <div id="footer-wrapper">
            <img src={logoSmall} alt="company-logo-small"/>
            <div>
                <ul>
                <li>Contact Us</li>
                <li>Help Center</li>
                <li>The Group</li>
                <li>Affliates</li>
                </ul>
            </div>
            <div>
                <ul>
                <li>Information</li>
                <li>Terms & Conditions</li>
                <li>Payment methods</li>
                <li>Bonus Terms</li>
                <li>Responsible Gaming</li>
                </ul>
            </div>
            </div>
        </footer>
    )
}