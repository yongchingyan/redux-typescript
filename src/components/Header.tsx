import React from 'react'
import '../css/header.css';
import logo from '../asset/logo.png';

export const Header = () => {
    return (
        <header id="header-wrapper">
            <img id="header-logo" src={logo} alt="company logo"/>
            <div id="header-auth">
            <button id="signup-button"> Sign up </button>
            <button id="login-button"> Login </button>
            </div>
            <div id="header-nav">
            <ul id="nav-games">
                <li>Home</li>
                <li>RPG</li>
                <li>Arcade</li>
                <li>Brain Games</li>
            </ul>
            <ul id="nav-support">
                <li>Support</li>
                <li>Contact Us</li>
            </ul>
            </div>
        </header>
    )
}