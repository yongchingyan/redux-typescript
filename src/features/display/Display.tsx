import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  setPC,
  setMobile,
  selectMode,
} from './displaySlice';

export var Display = () => {
    const mode = useSelector(selectMode);
    const dispatch = useDispatch();

    return (
        <div>
            <p>Current Mode: {mode}</p>
            <button onClick={()=> { 
                console.log("set mobile")
                dispatch(setMobile) 
            }}> Switch Mode </button>
        </div>
    )
}