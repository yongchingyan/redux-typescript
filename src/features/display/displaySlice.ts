import { createSlice } from '@reduxjs/toolkit';
import { RootState } from '../../app/store';

interface CounterState {
    mode: string;
}

const initialState: CounterState = {
    mode: "pc"
}

export const displaySlice = createSlice({
    name: 'display',
    initialState,
    reducers: {
        setPC: state => { state.mode = "pc" },
        setMobile: state => { state.mode = "mobile" },
    },
});

export const {setPC, setMobile} = displaySlice.actions;

export const selectMode = (state: RootState) => state.display.mode;

export default displaySlice.reducer;