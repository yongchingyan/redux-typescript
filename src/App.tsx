import React from 'react';
import { Display } from './features/display/Display';
import { Header } from './components/Header';
import { Footer } from './components/Footer';
import { MainContent } from './components/Content';

function App() {
  return (
    <div className="App">
      <Header />
      <Display />
      <MainContent />
      <Footer />
    </div>
  );
}

export default App;