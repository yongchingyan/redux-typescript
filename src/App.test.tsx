import React from 'react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './app/store';
import App from './App';
import { Header } from './components/Header'
import { Display } from './features/display/Display'


test('Check if login buttons and navbar is rendered', () => {
  const { getByText } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );

  expect(getByText(/Login/i)).toBeInTheDocument();
  expect(getByText(/Home/i)).toBeInTheDocument();
});

test('Display should switch to mobile mode with windows innerwidth changes to below 1000px', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Display />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(()=> {
    // set innerwidth to
    global.innerWidth = 500;
    // Trigger the window resize event.
    global.dispatchEvent(new Event('resize')); 
  })

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


