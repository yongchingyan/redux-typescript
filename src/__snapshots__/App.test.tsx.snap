// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should switch to mobile mode with windows innerwidth changes to below 1000px 1`] = `
<div
  id="display"
>
  Current Display Mode: 
  pc
</div>
`;

exports[`Display should switch to mobile mode with windows innerwidth changes to below 1000px 2`] = `
<div
  id="display"
>
  Current Display Mode: 
  pc
</div>
`;
